### PR TITLE
Added parameters, scatterplot code, tests, updated docstring

### LINKED ISSUE #18: Code for Scatter plot

### PR DESCRIPTION
**Pull request by iamMoid**
Added new optional input parameters for added functionality, added test cases, added the scatterplot codes for different cases, and updated the docstring

### CODE REVIEW

File: src/magmaviz/scatterplot.py [hunk 1]
```diff
@@ -21,5 +54,144 @@ def scatterplot(df, x, y, c=""):
...
+            t = f"{xtitle.title()} vs {ytitle.title()}"
+        else:
+            t = f"{xtitle.title()} vs {ytitle.title()} by {ctitle.title()}"
+
+    # scatterplot
+    if c == "":
+        plot = alt.Chart(
+            data=df, title=t
+        ).mark_point(
+            opacity=o, size=s, color="purple"
```
Comment (miyer26) [inline src/magmaviz/scatterplot.py]: Just curious, would setting the color to purple here override the magma theme?

**Review by miyer26 — COMMENTED** [review]
Overall, the tests are comprehensive and sound. I just had one small comment, please take a look at it.

File: src/magmaviz/scatterplot.py (hunk 1 above)
Comment (iamMoid) [reply src/magmaviz/scatterplot.py]: Oh yes, good catch. I wanted to have a placeholder color and later update it based on the magma theme. Will update it soon.

Comment (miyer26) [reply src/magmaviz/scatterplot.py]: Sounds good!

**Review by miyer26 — APPROVED** [review]
Everything looks good!

File: src/magmaviz/scatterplot.py [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
+             Title of the color legend. Default value is blank. If not provided,
+             title will be proper case of the color column
+    xzero : boolean
+            Scale the x-axis to start from 0 by specifying True
+            Default value is set to False
+    yzero : boolean
+            Scale the y-axis to start from 0 by specifying True
+            Default value is set to False
+    shapes : boolean
+             Assigns the color column to shapes attribute of the plot if True
```
Comment (shyan0903) [inline src/magmaviz/scatterplot.py]:
```suggestion
             Assign the color column to the shape attribute of the plot if True
```

File: src/magmaviz/scatterplot.py [hunk 3]
```diff
@@ -1,17 +1,50 @@
...
+             Default value is set to False
```
Comment (shyan0903) [inline src/magmaviz/scatterplot.py]:
```suggestion
             Default value is set to True
```

**Review by shyan0903 — CHANGES_REQUESTED** [review]
Thank you for your great work! Please take a look at the suggested changes to your docstring parameter section and address the changes. Then it will be ready to be merged!

**Pull request merged**